Tools/TestRunnerShared/TestFeatures.cpp

```diff
@@ -106,15 +107,28 @@ static std::optional<double> overrideDeviceScaleFactorForTest(const std::string&
 
 static bool shouldDumpJSConsoleLogInStdErr(const std::string& pathOrURL)
 {
-    return pathContains(pathOrURL, "localhost:8800/beacon") || pathContains(pathOrURL, "localhost:9443/beacon")
-        || pathContains(pathOrURL, "localhost:8800/cors") || pathContains(pathOrURL, "localhost:9443/cors")
-        || pathContains(pathOrURL, "localhost:8800/fetch") || pathContains(pathOrURL, "localhost:9443/fetch")
-        || pathContains(pathOrURL, "localhost:8800/service-workers") || pathContains(pathOrURL, "localhost:9443/service-workers")
-        || pathContains(pathOrURL, "localhost:8800/streams/writable-streams") || pathContains(pathOrURL, "localhost:9443/streams/writable-streams")
-        || pathContains(pathOrURL, "localhost:8800/streams/piping") || pathContains(pathOrURL, "localhost:9443/streams/piping")
-        || pathContains(pathOrURL, "localhost:8800/xhr") || pathContains(pathOrURL, "localhost:9443/xhr")
-        || pathContains(pathOrURL, "localhost:8800/webrtc") || pathContains(pathOrURL, "localhost:9443/webrtc")
-        || pathContains(pathOrURL, "localhost:8800/websockets") || pathContains(pathOrURL, "localhost:9443/websockets");
+    auto url = URL({ }, String::fromUTF8(pathOrURL.c_str()));
+    if (!url.isValid())
+        return false;
+
+    auto host = url.host();
+    if (host != "localhost"_s && host != "web-platform.test"_s)
+        return false;
+
+    auto port = url.port().value_or(0);
+    if (port != 8800 && port != 9443)
+        return false;
+
+    auto path = url.path();
+    return path.contains("/beacon"_s)
+        || path.contains("/cors"_s)
+        || path.contains("/fetch"_s)
+        || path.contains("/service-workers"_s)
+        || path.contains("/streams/writable-streams"_s)
+        || path.contains("/streams/piping"_s)
+        || path.contains("/xhr"_s)
+        || path.contains("/webrtc"_s)
+        || path.contains("/websockets"_s);
```

Comment:
Can we not get away with just having in TestExpectations:

```
imported/w3c/web-platform-tests/beacon [ DumpJSConsoleLogInStdErr ]
```

etc.? Doesn't the inheritance that TestExpectations does work here?

Reply:
I don't think we should use TestExpectations to control test behavior. Let's leave it for documenting whether a test is broken or not.

Reply:
Well, we do support `DumpJSConsoleLogInStdErr` there already. I'd rather we configure it in fewer places than more and having test behavior documented in `Tools/` does seem suboptimal.

Reply:
We could probably update the inheritance but it does not work this way yet unfortunately, say you have:
imported/w3c/web-platform-tests/websockets [DumpJSConsoleLogInStdErr]
imported/w3c/web-platform-tests/websockets/test.html [Pass Failure]

test.html will not have the DumpJSConsoleLogInStdErr flag on.

Reply:
Ah I was wondering about that later on. That sucks.

Reply:
Ah, I thought the inheritance should make that work. That seems like something we should file a separate bug for.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=256895